packages/eas-cli/src/project/metroConfig.ts

```diff
@@ -0,0 +1,43 @@
+import * as MetroConfig from '@expo/metro-config';
+import { exit } from '@oclif/errors';
+import chalk from 'chalk';
+
+import { BuildContext } from '../build/context';
+import { Platform } from '../build/types';
+import Log, { learnMore } from '../log';
+import { confirmAsync } from '../prompts';
+
+export async function validateMetroConfigForManagedWorkflowAsync(ctx: BuildContext<Platform>) {
+  if (!(await MetroConfig.existsAsync(ctx.projectDir))) {
+    return;
+  }
+
+  const metroConfig = await MetroConfig.loadAsync(ctx.projectDir, { withoutDefaults: true });
```

Comment:
i believe this is the most correct, but it does add to the size of eas-cli. i think we can handle this validation without depending on `@expo/metro-config` by doing copying over this code: https://github.com/expo/expo-cli/blob/1cb14f0989dc8b778f66bafe25758fde31c6b138/packages/metro-config/src/ExpoMetroConfig.ts#L237-L248

the only thing to be careful of is metro.config.js can export an async function or an object, so we should evaluate it if it's an async function before inspecting the output